google/cloud/bigquery/_helpers.py

```diff
@@ -309,6 +309,46 @@ def _json_from_json(value, field):
         return None
 
 
+def _range_element_from_json(value, field):
+    """Coerce 'value' to a range element value, if set or not nullable."""
+    if value == "UNBOUNDED":
+        return None
+    elif field.element_type == "DATE":
```

Comment:
Do we have a list of the supported range element types somewhere?  If so, this could like be simplified to something like
```
if field.element_type IN _SUPPORTED_RANGE_ELEMENTS:
  return _CELLDATA[field.element_type](value, None)
else:
  raise ValueError...
```

Reply:
This way looks so much neater! I updated the code.